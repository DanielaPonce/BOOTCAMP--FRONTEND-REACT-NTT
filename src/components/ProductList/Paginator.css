.paginator-container {
	display: flex;
	align-items: center;
	column-gap: 16px;
}

.paginator-container button {
	box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
	background-color: var(--white);
	border-radius: 50%;
    border: unset;
	padding: 4px 8px;
	cursor: pointer;
    transition: all 0.3s;
}

.paginator-container button:hover {
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.3);
}