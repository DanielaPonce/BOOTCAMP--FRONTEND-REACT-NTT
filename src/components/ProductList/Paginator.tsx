import './Paginator.css';

type PaginatorProps = {
	currentPage: number;
	totalPages: number;
	onPageChange: (page: number) => void;
};

const Paginator: React.FC<PaginatorProps> = ({
	currentPage,
	totalPages,
	onPageChange
}) => {
	return (
		<div className="paginator-container">
			<button
				onClick={() => onPageChange(currentPage - 1)}
				disabled={currentPage === 1}
			>
				&lt;
			</button>
			<div id="paginator-content__number">
				Page {currentPage} of {totalPages}
			</div>
			<button
				onClick={() => onPageChange(currentPage + 1)}
				disabled={currentPage === totalPages}
			>
				&gt;
			</button>
		</div>
	);
};

export default Paginator;
