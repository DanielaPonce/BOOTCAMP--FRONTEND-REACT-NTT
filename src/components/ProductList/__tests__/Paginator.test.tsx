import { fireEvent, render, screen } from '@testing-library/react';
import Paginator from '../Paginator';

describe('Paginator Component', () => {
	const mockOnPageChange = jest.fn();

	beforeEach(() => {
		jest.clearAllMocks();
	});

	test('renders correctly', () => {
		render(
			<Paginator
				currentPage={1}
				totalPages={5}
				onPageChange={mockOnPageChange}
			/>
		);

		const prevButton = screen.getByRole('button', { name: '<' });
		const nextButton = screen.getByRole('button', { name: '>' });
		const pageInfo = screen.getByText('Page 1 of 5');

		expect(prevButton).toBeInTheDocument();
		expect(nextButton).toBeInTheDocument();
		expect(pageInfo).toBeInTheDocument();
	});

	test('disables the previous button on the first page', () => {
		render(
			<Paginator
				currentPage={1}
				totalPages={5}
				onPageChange={mockOnPageChange}
			/>
		);

		const prevButton = screen.getByRole('button', { name: '<' });
		expect(prevButton).toBeDisabled();
	});

	test('disables the next button on the last page', () => {
		render(
			<Paginator
				currentPage={5}
				totalPages={5}
				onPageChange={mockOnPageChange}
			/>
		);

		const nextButton = screen.getByRole('button', { name: '>' });
		expect(nextButton).toBeDisabled();
	});

	test('calls onPageChange with the correct value when navigating', () => {
		render(
			<Paginator
				currentPage={3}
				totalPages={5}
				onPageChange={mockOnPageChange}
			/>
		);

		const prevButton = screen.getByRole('button', { name: '<' });
		const nextButton = screen.getByRole('button', { name: '>' });

		fireEvent.click(prevButton);
		expect(mockOnPageChange).toHaveBeenCalledTimes(1);
		expect(mockOnPageChange).toHaveBeenCalledWith(2);

		fireEvent.click(nextButton);
		expect(mockOnPageChange).toHaveBeenCalledTimes(2);
		expect(mockOnPageChange).toHaveBeenCalledWith(4);
	});

	test('matches snapshot', () => {
		const { asFragment } = render(
			<Paginator
				currentPage={3}
				totalPages={5}
				onPageChange={mockOnPageChange}
			/>
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
