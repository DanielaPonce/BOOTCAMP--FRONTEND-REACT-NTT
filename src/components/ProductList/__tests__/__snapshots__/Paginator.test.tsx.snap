// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="paginator-container"
  >
    <button>
      &lt;
    </button>
    <div
      id="paginator-content__number"
    >
      Page 3 of 5
    </div>
    <button>
      &gt;
    </button>
  </div>
</DocumentFragment>
`;
